pkg/database/realm.go

```diff
@@ -1981,17 +2002,17 @@ func (db *Database) AllRealmCodeStats(ctx context.Context, requiredRealms int, e
 		) d
 		LEFT JOIN realm_stats s ON s.date = d.date
 		`
-	if realm != nil {
+	if len(excludedRealmIDs) > 0 {
 		// This needs the newlines in it
-		sql = sql + `WHERE s.realm_id != $3
+		sql = sql + `WHERE NOT s.realm_id = ANY($3)
```

Comment:
I think you might want `NOT IN` instead of `ANY`? I forget which is more performant.

Reply:
`WHERE s.realm_id NOT IN $3` gives a sql error

When $3 is an empty array - which could very well happen

```
{"severity":"DEBUG","timestamp":"2022-09-12T18:11:52.690161-07:00","logger":"database","message":"SELECT d.date AS date, COUNT(DISTINCT(realm_id)) as realm_id, SUM(s.codes_issued) AS codes_issued, SUM(s.codes_claimed) AS codes_claimed, 0 AS codes_invalid, 0 AS user_reports_issued, 0 AS user_reports_claimed, 0 AS tokens_claimed, 0 AS tokens_invalid, 0 AS user_report_tokens_claimed, array[]::integer[] AS code_claim_age_distribution, 0 AS code_claim_mean_age, array[0,0,0]::bigint[] AS codes_invalid_by_os, 0 AS user_reports_invalid_nonce FROM ( SELECT date::date FROM generate_series($1, $2, '1 day'::interval) date ) d LEFT JOIN realm_stats s ON s.date = d.date WHERE s.realm_id NOT IN ($3) GROUP BY d.date ORDER BY date DESC","build_id":"unknown","build_tag":"unknown","caller":"pkg/database/realm.go:2009","duration":0.0079645,"values":["2022-06-15T00:00:00Z","2022-09-13T00:00:00Z","'{}'"],"rows":0}
{"severity":"ERROR","timestamp":"2022-09-12T18:11:52.690191-07:00","caller":"controller/controller.go:80","message":"internal error","request_id":"b589fa51-5da8-4452-a55e-fc2e10ea2023","error":"pq: invalid input syntax for type integer: \"{}\""
```